File: exercism/java/hamming/src/main/java/Hamming.java
import java.util.stream.IntStream;

public class Hamming {
    final String leftStrand;
    final String rightStrand;

    public Hamming(String leftStrand, String rightStrand) {
        if(leftStrand.isEmpty() && !rightStrand.isEmpty()){
            throw new IllegalArgumentException("left strand must not be empty.");
        }
        if(rightStrand.isEmpty() && !leftStrand.isEmpty()){
            throw new IllegalArgumentException("right strand must not be empty.");
        }
        if (leftStrand.length() != rightStrand.length()) {
            throw new IllegalArgumentException("leftStrand and rightStrand must be of equal length.");
        }
        this.leftStrand = leftStrand;
        this.rightStrand = rightStrand;
    }

    public int getHammingDistance() {
        return (int) IntStream.range(0, leftStrand.length()).filter(i -> leftStrand.charAt(i) != rightStrand.charAt(i)).count();
    }
}

File: exercism/kotlin/hamming/src/main/kotlin/Hamming.kt
import java.util.stream.IntStream

object Hamming {

    fun compute(leftStrand: String, rightStrand: String): Int {
        if(leftStrand.length != rightStrand.length){
            throw IllegalArgumentException("left and right strands must be of equal length")
        }
        return IntStream.range(0, leftStrand.length)
            .filter({i -> !leftStrand.substring(i, i+1).equals(rightStrand.substring(i, i+1))})
            .count().toInt()
    }
}
